server/server_test.go

```diff
@@ -436,6 +436,49 @@ func TestServiceRegistryInfoWithNilTags(t *testing.T) {
 	test.Assert(t, drCount == 1)
 }
 
+func TestServiceRegistryInfoSkipListenAddr(t *testing.T) {
+	realAddr := utils.NewNetAddr("tcp", "127.0.0.1:9999")
+	registryInfo := registry.Info{
+		Weight:         100,
+		Addr:           realAddr,
+		SkipListenAddr: true,
+	}
+	checkInfo := func(info *registry.Info) {
+		test.Assert(t, info.Weight == registryInfo.Weight, info.Weight)
+		test.Assert(t, info.SkipListenAddr, info.SkipListenAddr)
+		test.DeepEqual(t, info.Addr, realAddr)
```

Comment:
What's the purpose of testing with lAddr == registryInfo.Addr == realAddr?

Reply:
If SkipListenAddr is true, then info.Addr will not be overwritten by laddr and will be equal to realAddr. Otherwise,  info.Addr will be overwritten by laddr and will not be equal to realAddr. info.Addr is **not** always equal to realAddr.